shm/src/Game.cpp

```diff
@@ -109,10 +102,52 @@ void Game::selectOption() {
             sell();
             break;
         case MenuOption::Exit :
-            exit(0);
+            if (exitGame() == false) {
+                menuOption_ = MenuOption::NoChoose;
+            }
+            break;
         default:
             std::cout << "Option doesn't exists\n";
     }
+    return menuOption_;
+}
+
+bool Game::validatingMenuChoose(size_t option) {
+    size_t firstOptionElement { 1 };
+    size_t lastOptionElement { 6 };
+    if (std::cin.fail() || option < firstOptionElement || option > lastOptionElement) {
+        std::cin.clear();
+        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+        std::cout << "Wrong value please insert number between " << firstOptionElement << " and " << lastOptionElement << ".\n";
+        return false;
+    }
+    return true;
+}
+
+Game::ConfirmOption Game::confirmOption(std::string announcemen) {
```

Comment:
method name 'confirmOption' very similar to enum 'ConfirmOpton' which is confusing - a better name would be 'checkAnswer'